### PR TITLE
RFC: Webview Bindings

### PR DESCRIPTION
**Pull request by nothingismagick**
Please consider the following RFC and leave your comments below.

https://github.com/tauri-apps/rfcs/blob/RFC_001/texts/0000-webview_bindings.md

### CODE REVIEW

**Comment by Akryum** [comment]
An audit of the current upstream repositories could steer the decision towards working with upstream or hard-forking. Is it currently difficult to make those upstream projects to move forward based on Tauri's and its users needs?

**Comment by nothingismagick** [comment]
There is currently a bit of momentum from Zserge to create an organization to host the core library as well as the bindings to rust, python etc. Hopefully, core members of Tauri can participate in this org as co-maintainers.

We have audited both codebases (the headers in Zserge and the ones in Boscop) and Boscop's headers have diverged from Zserge's to a moderate degree, and furthermore Zserge's have been modernized and are a great deal cleaner.

In an ideal world we would all work together, but at the end of the day the Tauri org has to be in a position to guarantee that neither upstream burnout nor bus-factor can put this project in jeopardy.

**Comment by mosheduminer** [comment]
One of the advantages of open source, I think, is the pooling of resources, so everyone can work on the same code base, instead of duplicate efforts.

If that ideal is not being reached (due to difficulties contributing, etc.) then I think it is preferable to fork.

(I would be willing to assist, but my experience is mainly with interpreted languages, so I don't know if there is anything I could help with).

**Comment by khionu** [comment]
Part of our user base will be larger companies that can't switch to products that have potential reliability issues. I agree with the OSS sentiments to a strong degree, but at the same time, we can't afford to compromise our reliability.

**Comment by nothingismagick** [comment]
So we had a group call yesterday, and discussed the following - which is in line with the RFC.

1. Fork webview-x headers to tauri-apps/webview
2. Start working on our own implementation of rust bindings to tauri-apps/webview
3. Setup tauri with a config that allows devs to choose from a different source for the headers and/or bindings.

If and when the `webview` organisation is founded, we can then offer our work as something compliant with the "official" headers.

@tensor-programming @lucasfernog @jbolda @khionu

**Comment by khionu** [comment]
SGTM!!

File: texts/0000-webview_bindings.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# RFC #0000 Webview
+
+| Status        | Proposed       |
+:-------------- | ---------------------------------------------------- |
+| **RFC #**     | [0000](https://github.com/tauri-apps/governance-and-guidance/pull/1)
```
Comment (khionu) [inline texts/0000-webview_bindings.md]: Formatting: missing trailing pipe

File: texts/0000-webview_bindings.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+| **Authors** | Daniel Thompson-Yvetot (denjell@tauri.studio), Tensor Programming (tensor@tauri.studio) |
+| **External Sponsor** | Erlend (e.soghe@gmail.com) |
+| **Updated**   | 2020-03-10                                           |
```
Comment (khionu) [inline texts/0000-webview_bindings.md]: Formatting: should remove excess whitespace or add it to the rest of the lines.

File: texts/0000-webview_bindings.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+We will create generic mappings to the rust bindings, this will allow developers to more easily switch to other forks of webview and potentially other rendering engines entirely.
+
+### Ideal Situation
+We become co-maintainers of the official webview/webview-rs bindings together with @boscop and consume official monolithic webview-x type headers directly from webview/webview
+
+### Acceptable Situation
+We become co-maintainers of webview/webview-rs and consume official monolithic webview-x type headers directly from webview/webview.
+
+### Backup Plan
+We consume webview-x type headers and bind them to tauri-apps/tauri-webview, essentially hard-forking. Contributions back upstream and manual merges from upstream possible, but will require additional effort (and attention).
```
Comment (khionu) [inline texts/0000-webview_bindings.md]: This is now our primary plan

**Comment by khionu** [comment]
Made some formatting fixes, as well as refactored the solution to reflect more recent developments.

**Pull request merged**